New Microsoft.Insights api version 2021-02-01-preview

avjerafi opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
Duplicated existing 2020-05-01-preview folder as the base for 2021-02-01-preview

Commit:
Applied changes for 2021-02-01-preview

Commit:
Added package tag in readme.md

chiragg4u commented on specification/monitor/resource-manager/Microsoft.Insights/preview/2021-02-01-preview/examples/listOperations.json:
```diff
@@ -0,0 +1,23 @@
+      "body": {
+        "value": [
+          {
+            "name": "Microsoft.Storage/scheduledQueryRules/write",
```
>"name": "Microsoft.Storage/scheduledQueryRules/write", [](start = 12, length = 54)

Examples should contain all the operations as these are used for documentation generation also. Please add other operations as well.

avjerafi replied:
Done.

Commit:
Added operations to example payload

Commit:
Fixed validation

Commit:
Fixed prettier for listOperations

Commit:
Removed operation endpoint

avjerafi commented:
Removed the Operations endpoint I've added, because it turns out it is already implemented by another service within Microsoft.Insights provider, at:
https://github.com/Azure/azure-rest-api-specs/tree/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01

Commit:
Appliod prettier fixes

avjerafi commented:
Please address the explanation I gave here earlier regarding the absence of an operations endpoint: it already exists and maintained elsewhere, and no changes are required for this new api version that I'm adding. I cannot and should not add the operations endpoint here. Please override the LintDiff failure result. 

Please let me know if there is anything else blocking the approval of this PR that I should/can address. It looks to me that none of the errors listed in the "SDK azure-sdk-for-net" has anything to do with my changes.

weidongxu-microsoft commented:
@avjerafi

Usually, all related json should be grouped in a single tag for completeness. This way, the operation endpoint will exist and be detected.

Is there any reason why insights doing these kind of single json tag? Java SDK would need complete tag like "pacakge-2020-03" to function.

@msyyc @ArcturusZhang  Does Python or Go SDK has any comment on this kind of single json tag?

weidongxu-microsoft commented:
@avjerafi

If no objection from other SDK owner, I am OK with this.

Let me know the target date of this swagger (i.e. when service onboard and you need the spec merged).

ArcturusZhang commented:
> @avjerafi
> 
> Usually, all related json should be grouped in a single tag for completeness. This way, the operation endpoint will exist and be detected.
> 
> Is there any reason why insights doing these kind of single json tag? Java SDK would need complete tag like "pacakge-2020-03" to function.
> 
> @msyyc @ArcturusZhang Does Python or Go SDK has any comment on this kind of single json tag?
> 
> ```
> ### Tag: package-2021-02-preview
> 
> These settings apply only when `--tag=package-2021-02-preview` is specified on the command line.
> 
> input-file:
> - Microsoft.Insights/preview/2021-02-01-preview/scheduledQueryRule_API.json
> ```

Go SDK only releases a package with full functionalities in one RP, therefore we will not make a release on this tag.
I would like to say we need to add this new swagger to one of the composite tag (a new one or exsiting one), but we could defer that until we get a release request. FYI @avjerafi

avjerafi commented:
@weidongxu-microsoft The target date for the swagger change is today. Thank you.
